src/features/social/PlayerModal.tsx

```diff
@@ -190,18 +172,46 @@ export const PlayerDetails: React.FC<Props> = ({ player }) => {
     }
   };
 
-  const sendMessage = async (message: FarmInteraction) => {
-    const { data } = await postEffect({
+  const sendMessage = async (message: string) => {
+    const newMessage: Interaction = {
+      type: "chat",
+      message,
+      recipient: {
+        id: player.farmId,
+        tokenUri: tokenUriBuilder(player.clothing),
+        username: player.username ?? `#${player.farmId}`,
+      },
+      sender: {
+        id: farmId,
+        tokenUri: tokenUriBuilder(myClothing),
+        username: myUsername ?? `#${farmId}`,
+      },
+      createdAt: Date.now(),
+    };
+
+    // Optimistically update the messages
+    mutate(
+      (current) =>
+        mergePlayerData(current!, {
+          messages: [...(current?.data?.messages ?? []), newMessage],
+        }),
+      { revalidate: false },
+    );
```

Comment:
```suggestion
mutate(
      async (current) => {
        const { data: response } = await postEffect({
          effect: {
            type: "message.sent",
            recipientId: player.farmId,
            message,
          },
          transactionId: randomID(),
          token: token as string,
          farmId: farmId,
        });

        return mergePlayerData(current!, response);
      },
      {
        optimisticData: (_, current) =>
          mergePlayerData(current!, {
            messages: [...(current?.data?.messages ?? []), newMessage],
          }),
      },
    );
```

I sent multiple messages, and found that sometimes data endpoint being queried while messages were still in flight. The state gets a bit confused about what the `current` value is.

Looks like mutate can take an async function, and it will block the refresh while the mutation is in progress.